src/modules/RA/DataLayer/components/DataLayerTabbedForm.js

```diff
@@ -111,13 +110,32 @@ const DataLayerTabbedForm = ({
         <BooleanInput source="active_by_default" />
 
         <LongTextInput source="description" label="datalayer.form.description" />
-      </CustomFormTab>
 
-      <CustomFormTab label="datalayer.form.style" path="style">
+        <FormDataConsumer>
+          {({ formData }) =>
+            // Allow to initialize default value even if next tab is not yet loaded
+            formData.source && (
+            <TextInput
+              type="hidden"
+              label=""
+              source="layer_style"
+              defaultValue={() => ({
+                layer_style: getLayerStyleDefaultValue(randomColor, getShapeFromGeomType(geomType)),
+              })}
+            />
+            )
+          }
+        </FormDataConsumer>
+
+      </LazyFormTab>
+
+      <LazyFormTab label="datalayer.form.style" path="style">
+
         <StyleField
           source="layer_style"
           withSource="source"
           label="datalayer.form.styles.mainstyle"
+          defaultValue={getLayerStyleDefaultValue(randomColor, getShapeFromGeomType(geomType))}
```

Comment:
Maybe you can declare the default value and call it in this line and line 123
`const layerStyleDefaultValue = getLayerStyleDefaultValue(randomColor, getShapeFromGeomType(geomType))`

Reply:
There is side effect here as the source is set later so the default value is invalid. Here the value is set only when the tab is opened.